### PR TITLE
Reduce the animated pictures size to save memory and avoid deadlock

### PR DESCRIPTION
**Pull request by anxuae**
I noticed deadlock / memory error when animated option is True, after several sequences.
This fix reduce the size of pictures processed by the subprocesses pool.

Don't noticed deadlock anymore, but have to perform more tests.

### CODE REVIEW

File: pibooth/pictures/factory.py [hunk 1]
```diff
@@ -273,9 +273,9 @@ def add_text(self, text, font_name, color, align=CENTER):
         assert align in [self.CENTER, self.RIGHT, self.LEFT], "Unknown aligment '{}'".format(align)
         self._texts.append((text, fonts.get_filename(font_name), color, align))
         if self.is_portrait:
-            self._texts_height = 600
+            self._texts_height = int(self.height * 0.167)
```
Comment (werdeil) [inline pibooth/pictures/factory.py:276]: I would rather use  int(self.height / 6) here, is there a reason to use * 0.167 instead?

File: pibooth/pictures/factory.py [hunk 2]
```diff
@@ -273,9 +273,9 @@ def add_text(self, text, font_name, color, align=CENTER):
...
         else:
-            self._texts_height = 300
+            self._texts_height = int(self.height * 0.125)
```
Comment (werdeil) [inline pibooth/pictures/factory.py:278]: I would rather use int(self.height / 8) here, is there a reason to use * 0.125 instead?

File: pibooth/booth.py [hunk 3]
```diff
@@ -375,6 +376,10 @@ def main_loop(self):
 def main():
     """Application entry point.
     """
+    if hasattr(multiprocessing, 'set_start_method'):
+        # Avoid use 'fork': safely forking a multithreaded process is problematic
+        multiprocessing.set_start_method('spawn')
```
Comment (werdeil) [inline pibooth/booth.py:381]: I don't understand how this can have an effect, is the multiprocessing package used by pluggy?

**Review by werdeil — COMMENTED** [review]
Small questions for you @anxuae :-)

File: pibooth/pictures/factory.py (hunk 1 above)
Comment (anxuae) [reply pibooth/pictures/factory.py:276]: That's True

File: pibooth/pictures/factory.py (hunk 2 above)
Comment (anxuae) [reply pibooth/pictures/factory.py:278]: True

File: pibooth/booth.py (hunk 3 above)
Comment (anxuae) [reply pibooth/booth.py:381]: The multiprocessing module is used by the `pibooth/pictures/pool.py` to generate each frame of the animated picture in a different process, however the choice of the method used to fork the main process shall be protected by the `if __name__ == '__main__'` condition has explained in the documentation, that is why this is done at this step.

**Review by werdeil — APPROVED** [review]
Ok to merge

**Pull request merged**